### ISSUE TITLE
CoW IGO

### ISSUE THREAD

**Issue by vincentarelbundock** [cjyetman/memberstates#2]
Apparently there's an update coming at some point.

http://www.correlatesofwar.org/data-sets/IGOs/IGO_codebook_v2.1.pdf

**Comment by cjyetman** [comment]
Ugh... this forces me to recognize that this package will eventually have the same problem that countrycode has... these things are temporal.

**Comment by vincentarelbundock** [comment]
And my guess is it wouldnt be easy to scrape historical.mberships from original sources, unfortunately. 

Also, I haven't looked closely at the content of the cow dataset, but it seems like memberstates needs to define it's comparative advantage.

**Comment by cjyetman** [comment]
well, for one I envision something that defines membership broader than just membership in an IGO... so it can/does contain membership info for various categorizations of countries (economic, geographic, political, etc.), not just IGO membership

there's also something to be said for making information more easily accessible/usable... even if it simply repackaged data from the CoW IGO, a well-crafted user interface/API might be worthwhile

**Comment by vincentarelbundock** [comment]
Agreed.

**Comment by briatte** [comment]
> these things are temporal

Wouldn't it be an interesting development to code memberships with their start-end dates, e.g. UK in the EU, and/or to add a `version` argument that would accommodate for various versions of a dataset, e.g. COW? I understand this would make the package much, _much_ more difficult to develop and maintain, though.

**Comment by vincentarelbundock** [comment]
@briatte did you look at the CoW "IGO dataset"? This is pretty much exactly what they offer.

**Comment by cjyetman** [comment]
Yeah, I started this before @vincentarelbundock began with the timeseries project for `countrycode`, so it wasn't part of the original plan. Dealing with versions/temporal issues drastically increases the complexity. The goal here, and the primary value-added was to make getting a list of member countries for a given thing much easier, so to stick to that **and** add versions and variations based on year would require some serious thought. Open to suggestions...

**Comment by cjyetman** [comment]
I should also say that my goal was to plan this so everything was auto-scrpaed from an original/official source... that way maintenance of it would consist of me coming here every 6 months or so, running the build script, fixing any minor issues that have come up, and if there's anything different preparing a new version for release.

**Comment by briatte** [comment]
@vincentarelbundock I did not know about that part of COW — good to know!
